code/modules/food_and_drinks/food/foods/desserts.dm

```diff
@@ -19,7 +19,8 @@
 /obj/item/reagent_containers/food/snacks/icecream/update_overlays()
 	. = ..()
 	var/mutable_appearance/filling = mutable_appearance('icons/obj/kitchen.dmi', "icecream_color")
-	filling.color = mix_color_from_reagents(reagents.reagent_list)
+	var/list/reagent_colors = rgb2num(mix_color_from_reagents(reagents.reagent_list), COLORSPACE_HSV)  //switching to HSV colorspace lets us easily manipulate the saturation and brightness independently
+	filling.color = rgb(reagent_colors[1], ((1.5*reagent_colors[2])-10), (clamp(reagent_colors[3], 85, 100)-10), space = COLORSPACE_HSV) //Clamping the brightness keeps us from having greyish ice cream while still alowing for a range of colours
```

Comment:
```suggestion
	//Clamping the brightness keeps us from having greyish ice cream while still alowing for a range of colours
	filling.color = rgb(reagent_colors[1], ((1.5 * reagent_colors[2]) - 10), (clamp(reagent_colors[3], 85, 100) - 10), space = COLORSPACE_HSV) 
```
two things
1. we can move the the comment to the line before to keep the line length readable
2. spacing so the math is more readable